WalletWasabi.Gui/Global.cs

```diff
@@ -737,30 +735,22 @@ public async Task DisposeAsync()
 					}
 				}
 
-				Logger.LogDebug($"Step #13: {nameof(TorMonitor)}.", nameof(Global));
-
-				if (TorMonitor is { } torMonitor)
-				{
-					await torMonitor.StopAsync().ConfigureAwait(false);
-					Logger.LogInfo($"{nameof(TorMonitor)} is stopped.");
-				}
-
-				Logger.LogDebug($"Step #14: {nameof(TorManager)}.", nameof(Global));
+				Logger.LogDebug($"Step: {nameof(TorManager)}.", nameof(Global));
 
 				if (TorManager is { } torManager)
```

Comment:
What came to my mind here is that we can have a problem with the order of disposal later. TorMonitor needs to be stopped before TorManager. This puts on the question if TorMonitor should handle the disposal of TorManager, moreover if TorMonitor cannot live without TorManager then TorMonitor could handle the life of TorManager at all.

Reply:
In my mind, `TorProcessManager` is something that manages Tor process. Meaning, it can start it, it can stop it and that's basically all. `TorMonitor` is a component that simply checks whether Tor process is running as expected. So it's built on top of `TorProcessManager`. This leads to the following correct way of starting and disposal:

```
TorProcessManager.Start
TorMonitor.Start
TorMonitorDispose
TorProcessManager.Dispose
```

And that's how it behaves with this PR as HostedServices are stopped (disposed) before TorProcessManager. 

With the current design, if you were to say, I don't want `TorMonitor`, you can simply disable it with one IF statement and it would work as one expects. So I find it flexible too.

Does that answer your concerns? Or did you ask something else?

Reply:
Yes, it answers. I would like to prevent the restart if TorProcessManager was disposed. How about adding isStopped or something like that and prevent TorProcessManager.StartAsync() in that case - So TorMonitor won't be able to restart Tor accidentally. Even later if someone messes up the disposal order.

Reply:
Just to clarify: Is it something that can happen now? Or do you talk only about possibility in the future?

Reply:
But I see your point.

Reply:
In the future.

Reply:
Please see https://github.com/zkSNACKs/WalletWasabi/pull/4648/commits/8deb46d635f2e02d72e191774dab234593898bcd